src/file_integrity.rs

```diff
@@ -0,0 +1,46 @@
+/// TODO: how to retrieve the actual pure checksum (reference value)
+use serde::{Deserialize, Serialize};
+
+/// Fictional struct for the moment / prob use an external library
+#[derive(PartialEq)]
+pub struct Checksum {
+    value: u32,
+}
+
+/// A MK8 game file and its expected `Checksum` (ref. value).
+#[derive(PartialEq)]
+struct GameFile {
+    path: &'static str,
+    expected_checksum: Checksum,
+}
+
+impl GameFile {
+    pub fn compute_real_checksum(&self) -> Checksum {
+        // TODO: compute real checksum using `self.path`
+        Checksum { value: 42 }
+    }
+
+    pub fn checksums_match(&self) -> bool {
+        self.expected_checksum == self.compute_real_checksum()
+    }
+}
+
+/// Are all those derive useful?
```

Comment:
note: This is a doc comment, so it renders into the documentation. You probably don't want that, so use two slashes: https://imgur.com/a/fiMEjpp

And, no, they're not currently useful. Like I said before, though, you'll want most of your types to implement traits "all the way down." A type can't be `Clone` unless its children are, nor can it be `PartialEq` or `Deserialize`.